### ISSUE TITLE
Proposal: Allow 3-digit-grouped binary in `non_octal_unix_permissions`

### ISSUE THREAD

**Issue by SabrinaJewson** [rust-lang/rust-clippy#11243]
### Description

I personally much prefer to write Unix permissions in binary rather than octal. For comparison:

| Format | Representation |
| --- | --- |
| ls(1) | `rwxr-xr--` |
| Binary | `0b111_101_100` |
| Octal | `0o754` |
| chmod(1) | `ugo+rwx,go-w,o-x` or `u+rwx,g+rx-w,o+r-wx` |

The 3-group binary format is more readable than octal in my opinion, because it lines up very neatly with the ls(1)  `rwxrwxrwx` format, and does not require any binary conversion or lookup tables in one’s head.

Currently Clippy disallows the following code:

```rs
fn main() {
    std::fs::Permissions::from_mode(0b111_101_100);
}
use std::os::unix::prelude::PermissionsExt as _;
```

To support this pattern, Clippy should include an exception in the `non_octal_unix_permissions` and `unusual_byte_groupings` lints, to specifically allow binary literals of the exact format `0b[XXX_]XXX_XXX_XXX`.

### Version

```text
rustc 1.71.0 (8ede3aae2 2023-07-12)
binary: rustc
commit-hash: 8ede3aae28fe6e4d52b38157d7bfe0d3bceef225
commit-date: 2023-07-12
host: x86_64-unknown-linux-gnu
release: 1.71.0
LLVM version: 16.0.5
```

### Additional Labels

@rustbot label +C-enhancement

**Comment by Alexendoo** [comment]
It would be reasonable to plain allow binary literals in general

**Comment by Centri3** [comment]
I agree, in the case of bitflags of any kind it's much easier to understand using binary
